src/core/Akka.Cluster.Tests.MultiNode/SurviveNetworkInstabilitySpec.cs

```diff
@@ -0,0 +1,555 @@
+                            () => Sys.ActorOf<SurviveNetworkInstabilitySpecConfig.Echo>()).ToImmutableHashSet();
+
+                    Sys.ActorSelection(Node(_config.Third) / "user" / "watcher").Tell(new SurviveNetworkInstabilitySpecConfig.Targets(refs));
+                    ExpectMsg<SurviveNetworkInstabilitySpecConfig.TargetsRegistered>();
+                }, _config.Second);
+
+                EnterBarrier("targets-registered");
+
+                RunOn(() =>
+                {
+                    foreach (var role in others)
+                    {
+                        TestConductor.Blackhole(role, _config.Second, ThrottleTransportAdapter.Direction.Both).Wait();
+                    }
+                }, _config.First);
+                EnterBarrier("blackhole-6");
+
+                RunOn(() =>
+                {
+                    // this will trigger watch of targets on second, resulting in too many outstanding
+                    // system messages and quarantine
+                    Sys.ActorSelection("/user/watcher").Tell("boom");
+                    Within(10.Seconds(), () =>
+                    {
+                        ExpectMsg<QuarantinedEvent>().Address.Should().Be(GetAddress(_config.Second));
+                    });
+                    Sys.EventStream.Unsubscribe(TestActor, typeof(QuarantinedEvent));
+                }, _config.Third);
+                EnterBarrier("quarantined");
+
+                RunOn(() =>
+                {
+                    // second should be removed because of quarantine
+                    AwaitAssert(() => ClusterView.Members.Select(c => c.Address).Should().NotContain(GetAddress(_config.Second)));
+                }, others.ToArray());
+
+                EnterBarrier("after-6");
+                AssertCanTalk(others.ToArray());
+            });
+        }
+
+        public void A_Network_partition_tolerant_cluster_must_continue_and_move_Joining_to_Up_after_downing_of_one_half()
+        {
+            Within(60.Seconds(), () =>
+            {
+                // note that second is already removed in previous step
+                var side1 = ImmutableArray.Create(_config.First, _config.Third, _config.Fourth);
+                var side1AfterJoin = side1.Add(_config.Eighth);
+                var side2 = ImmutableArray.Create(_config.Fifth, _config.Sixth, _config.Seventh);
+
+                RunOn(() =>
+                {
+                    foreach (var role1 in side1AfterJoin)
+                    {
+                        foreach (var role2 in side2)
+                        {
+                            TestConductor.Blackhole(role1, role2, ThrottleTransportAdapter.Direction.Both).Wait();
+                        }
+                    }
+                }, _config.First);
+                EnterBarrier("blackhole-7");
+
+                RunOn(() =>
+                {
+                    AssertUnreachable(side2.ToArray());
```

Comment:
So is this spec completed now or are you still working on sorting out some issues with it @alexvaluyskiy ?